include/fixed_containers/fixed_bitset.hpp

```diff
@@ -0,0 +1,672 @@
+    {
+        validate(pos);
+        return subscript(pos);
+    }
+
+    constexpr reference operator[](size_t pos)
+    {
+        validate(pos);
+        return reference(*this, pos);
+    }
+
+    constexpr FixedBitset() noexcept
+      : data()
+    {
+    }  // construct with all false values
+
+    static constexpr bool NEED_MASK = BIT_COUNT < CHAR_BIT * sizeof(UnsignedLongLong);
+
+    static constexpr UnsignedLongLong MASK = (1ULL << (NEED_MASK ? BIT_COUNT : 0)) - 1ULL;
+
+    constexpr FixedBitset(UnsignedLongLong val) noexcept
+      : data{static_cast<Ty>(NEED_MASK ? val & MASK : val)}
+    {
+    }
+
+    template <class Traits, class Elem>
+    constexpr void construct(const Elem* const ptr,
+                             size_t count,
+                             const Elem elem0,
+                             const Elem elem1)
+    {
+        if (count > BIT_COUNT)
+        {
+            for (size_t idx = BIT_COUNT; idx < count; ++idx)
+            {
+                const auto character = ptr[idx];
+                if (!Traits::eq(elem1, character) && !Traits::eq(elem0, character))
+                {
+                    x_inv();
+                }
+            }
+
+            count = BIT_COUNT;
+        }
+
+        size_t w_pos = 0;
+        if (count != 0)
+        {
+            size_t bits_used_in_word = 0;
+            auto last = ptr + count;
+            Ty this_word = 0;
+            do
+            {
+                --last;
+                const auto character = *last;
+                this_word |= static_cast<Ty>(Traits::eq(elem1, character)) << bits_used_in_word;
+                if (!Traits::eq(elem1, character) && !Traits::eq(elem0, character))
+                {
+                    x_inv();
+                }
+
+                if (++bits_used_in_word == BITS_PER_WORD)
+                {
+                    data[w_pos] = this_word;
+                    ++w_pos;
+                    this_word = 0;
+                    bits_used_in_word = 0;
+                }
+            } while (ptr != last);
+
+            if (bits_used_in_word != 0)
+            {
+                data[w_pos] = this_word;
+                ++w_pos;
+            }
+        }
+
+        for (; w_pos <= WORD_COUNT; ++w_pos)
+        {
+            data[w_pos] = 0;
+        }
+    }
+
+    template <class Elem, class Traits, class Alloc>
+    constexpr explicit FixedBitset(
+        const std::basic_string<Elem, Traits, Alloc>& str,
+        typename std::basic_string<Elem, Traits, Alloc>::size_type pos = 0,
+        typename std::basic_string<Elem, Traits, Alloc>::size_type count =
+            std::basic_string<Elem, Traits, Alloc>::npos,
+        Elem elem0 = static_cast<Elem>('0'),
+        Elem elem1 = static_cast<Elem>('1'))
+    {
+        // construct from [_Pos, _Pos + _Count) elements in string
+        if (str.size() < pos)
+        {
+            x_ran();  // _Pos off end
+        }
```

Comment:
Why do we check manually here with an opaque "x_ran" function for errors when we also have a `validate` function above that can do this?

Reply:
Replaced with `preconditions` like in the rest of the library.